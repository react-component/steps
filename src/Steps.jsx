/* eslint react/no-did-mount-set-state: 0 */
import React, { cloneElement, Children, Component } from 'react';
import PropTypes from 'prop-types';
import { findDOMNode } from 'react-dom';
import classNames from 'classnames';
import debounce from 'lodash/debounce';
import { isFlexSupported } from './utils';

export default class Steps extends Component {
  static propTypes = {
    prefixCls: PropTypes.string,
    className: PropTypes.string,
    iconPrefix: PropTypes.string,
    direction: PropTypes.string,
    labelPlacement: PropTypes.string,
    children: PropTypes.any,
    status: PropTypes.string,
    size: PropTypes.string,
    progressDot: PropTypes.oneOfType([
      PropTypes.bool,
      PropTypes.func,
    ]),
    style: PropTypes.object,
    initial: PropTypes.number,
    current: PropTypes.number,
    icons: PropTypes.shape({
      finish: PropTypes.node,
      error: PropTypes.node,
    }),
    onChange: PropTypes.func,
  };
  static defaultProps = {
    prefixCls: 'rc-steps',
    iconPrefix: 'rc',
    direction: 'horizontal',
    labelPlacement: 'horizontal',
    initial: 0,
    current: 0,
    status: 'process',
    size: '',
    progressDot: false,
  };
  constructor(props) {
    super(props);
    this.state = {
      flexSupported: true,
      lastStepOffsetWidth: 0,
    };
    this.calcStepOffsetWidth = debounce(this.calcStepOffsetWidth, 150);
  }
  componentDidMount() {
    this.calcStepOffsetWidth();
    if (!isFlexSupported()) {
      this.setState({
        flexSupported: false,
      });
    }
  }
  componentDidUpdate() {
    this.calcStepOffsetWidth();
  }
  componentWillUnmount() {
    if (this.calcTimeout) {
      clearTimeout(this.calcTimeout);
    }
    if (this.calcStepOffsetWidth && this.calcStepOffsetWidth.cancel) {
      this.calcStepOffsetWidth.cancel();
    }
  }

  onStepClick = (current) => {
    const { onChange } = this.props;
    if (onChange) {
      onChange(current);
    }
  };

  calcStepOffsetWidth = () => {
    if (isFlexSupported()) {
      return;
    }
    // Just for IE9
    const domNode = findDOMNode(this);
    if (domNode.children.length > 0) {
      if (this.calcTimeout) {
        clearTimeout(this.calcTimeout);
      }
      this.calcTimeout = setTimeout(() => {
        // +1 for fit edge bug of digit width, like 35.4px
        const lastStepOffsetWidth = (domNode.lastChild.offsetWidth || 0) + 1;
        // Reduce shake bug
        if (this.state.lastStepOffsetWidth === lastStepOffsetWidth ||
          Math.abs(this.state.lastStepOffsetWidth - lastStepOffsetWidth) <= 3) {
          return;
        }
        this.setState({ lastStepOffsetWidth });
      });
    }
  }
  render() {
    const {
      prefixCls, style = {}, className, children, direction,
      labelPlacement, iconPrefix, status, size, current, progressDot, initial,
      icons, onChange,
      ...restProps,
    } = this.props;
    const { lastStepOffsetWidth, flexSupported } = this.state;
    const filteredChildren = React.Children.toArray(children).filter(c => !!c);
    const lastIndex = filteredChildren.length - 1;
    const adjustedlabelPlacement = !!progressDot ? 'vertical' : labelPlacement;
    const classString = classNames(prefixCls, `${prefixCls}-${direction}`, className, {
      [`${prefixCls}-${size}`]: size,
      [`${prefixCls}-label-${adjustedlabelPlacement}`]: direction === 'horizontal',
      [`${prefixCls}-dot`]: !!progressDot,
      [`${prefixCls}-flex-not-supported`]: !flexSupported,
    });

    return (
      <div className={classString} style={style} {...restProps}>
        {
          Children.map(filteredChildren, (child, index) => {
            if (!child) {
              return null;
            }
            const stepNumber = initial + index;
            const childProps = {
              stepNumber: `${stepNumber + 1}`,
              stepIndex: stepNumber,
              prefixCls,
              iconPrefix,
              wrapperStyle: style,
              progressDot,
              icons,
              onStepClick: onChange && this.onStepClick,
              ...child.props,
            };
            if (!flexSupported && direction !== 'vertical' && index !== lastIndex) {
              childProps.itemWidth = `${100 / lastIndex}%`;
              childProps.adjustMarginRight = -Math.round(lastStepOffsetWidth / lastIndex + 1);
            }
            // fix tail color
            if (status === 'error' && index === current - 1) {
              childProps.className = `${prefixCls}-next-error`;
            }
            if (!child.props.status) {
              if (stepNumber === current) {
                childProps.status = status;
              } else if (stepNumber < current) {
                childProps.status = 'finish';
              } else {
                childProps.status = 'wait';
              }
            }
            return cloneElement(child, childProps);
          })
        }
      </div>
    );
  }
}
