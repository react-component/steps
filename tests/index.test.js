import React from 'react';
import { render } from 'enzyme';
import Steps, { Step } from '../src';

describe('Steps', () => {
  describe('render', () => {
    const steps = (
      <Steps>
        <Step title="已完成" />
        <Step title="进行中" />
        <Step title="待运行" />
        <Step title="待运行" />
      </Steps>
    );

    it('renders correctly', () => {
      const wrapper = render(steps);
      expect(wrapper).toMatchSnapshot();
    });

    it('renders current correctly', () => {
      const wrapper = render(React.cloneElement(steps, { current: 2 }));
      expect(wrapper).toMatchSnapshot();
    });

    it('renders status correctly', () => {
      const wrapper = render(React.cloneElement(steps, { current: 2, status: 'error' }));
      expect(wrapper).toMatchSnapshot();
    });

    it('renders vertical correctly', () => {
      const wrapper = render(React.cloneElement(steps, { direction: 'vertical' }));
      expect(wrapper).toMatchSnapshot();
    });

    it('renders labelPlacement correctly', () => {
      const wrapper = render(React.cloneElement(steps, { labelPlacement: 'vertical' }));
      expect(wrapper).toMatchSnapshot();
    });

    it('renders progressDot correctly', () => {
      const wrapper = render(React.cloneElement(steps, { progressDot: true }));
      expect(wrapper).toMatchSnapshot();
    });

    it('renders progressDot function correctly', () => {
      const wrapper = render(React.cloneElement(steps, { progressDot: () => <span>a</span> }));
      expect(wrapper).toMatchSnapshot();
    });

    it('renders step with description', () => {
      const wrapper = render(
        <Steps>
          <Step title="已完成" description="xx" />
          <Step title="进行中" description="xx" />
          <Step title="待运行" description="xx" />
          <Step title="待运行" description="xx" />
        </Steps>
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('renders step with description and status', () => {
      const wrapper = render(
        <Steps>
          <Step title="已完成" description="xx" status="wait" />
          <Step title="进行中" description="xx" status="wait" />
          <Step title="待运行" description="xx" status="process" />
          <Step title="待运行" description="xx" status="finish" />
        </Steps>
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('renders with fasly children', () => {
      const wrapper = render(
        <Steps>
          <Step title="已完成" description="xx" status="wait" />
          <Step title="进行中" description="xx" status="wait" />
          {undefined}
          <Step title="待运行" description="xx" status="process" />
          {false}
          <Step title="待运行" description="xx" status="finish" />
          {null}
        </Steps>
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('renders step with tailContent', () => {
      const wrapper = render(
        <Steps>
          <Step title="已完成" description="xx" tailContent="text" />
          <Step title="进行中" description="xx" tailContent={<div>content</div>} />
          <Step title="待运行" description="xx" tailContent={3} />
          <Step title="待运行" description="xx" tailContent="text" />
        </Steps>
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('renders step with specified starting number', () => {
      const wrapper = render(React.cloneElement(steps, { start: 4 }));
      expect(wrapper).toMatchSnapshot();
    });
  });
});
