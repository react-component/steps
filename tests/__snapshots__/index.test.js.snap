// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps render renders correctly 1`] = `
<div
  class="rc-steps rc-steps-horizontal rc-steps-label-horizontal"
>
  <div
    class="rc-steps-item rc-steps-item-process"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        1
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        已完成
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        2
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        进行中
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        3
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        4
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps render renders current correctly 1`] = `
<div
  class="rc-steps rc-steps-horizontal rc-steps-label-horizontal"
>
  <div
    class="rc-steps-item rc-steps-item-finish"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon rcicon rcicon-check"
      >
        
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        已完成
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-finish"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon rcicon rcicon-check"
      >
        
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        进行中
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-process"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        3
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        4
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps render renders labelPlacement correctly 1`] = `
<div
  class="rc-steps rc-steps-horizontal rc-steps-label-vertical"
>
  <div
    class="rc-steps-item rc-steps-item-process"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        1
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        已完成
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        2
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        进行中
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        3
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        4
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps render renders progressDot correctly 1`] = `
<div
  class="rc-steps rc-steps-horizontal rc-steps-label-vertical rc-steps-dot"
>
  <div
    class="rc-steps-item rc-steps-item-process"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        <span
          class="rc-steps-icon-dot"
        >
          
        </span>
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        已完成
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        <span
          class="rc-steps-icon-dot"
        >
          
        </span>
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        进行中
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        <span
          class="rc-steps-icon-dot"
        >
          
        </span>
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        <span
          class="rc-steps-icon-dot"
        >
          
        </span>
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps render renders progressDot function correctly 1`] = `
<div
  class="rc-steps rc-steps-horizontal rc-steps-label-vertical rc-steps-dot"
>
  <div
    class="rc-steps-item rc-steps-item-process"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        <span>
          a
        </span>
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        已完成
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        <span>
          a
        </span>
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        进行中
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        <span>
          a
        </span>
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        <span>
          a
        </span>
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps render renders status correctly 1`] = `
<div
  class="rc-steps rc-steps-horizontal rc-steps-label-horizontal"
>
  <div
    class="rc-steps-item rc-steps-item-finish"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon rcicon rcicon-check"
      >
        
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        已完成
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-finish rc-steps-next-error"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon rcicon rcicon-check"
      >
        
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        进行中
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-error"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon rcicon rcicon-cross"
      >
        
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        4
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps render renders step with description 1`] = `
<div
  class="rc-steps rc-steps-horizontal rc-steps-label-horizontal"
>
  <div
    class="rc-steps-item rc-steps-item-process"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        1
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        已完成
      </div>
      <div
        class="rc-steps-item-description"
      >
        xx
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        2
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        进行中
      </div>
      <div
        class="rc-steps-item-description"
      >
        xx
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        3
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
      <div
        class="rc-steps-item-description"
      >
        xx
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        4
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
      <div
        class="rc-steps-item-description"
      >
        xx
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps render renders step with description and status 1`] = `
<div
  class="rc-steps rc-steps-horizontal rc-steps-label-horizontal"
>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        1
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        已完成
      </div>
      <div
        class="rc-steps-item-description"
      >
        xx
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        2
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        进行中
      </div>
      <div
        class="rc-steps-item-description"
      >
        xx
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-process"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        3
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
      <div
        class="rc-steps-item-description"
      >
        xx
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-finish"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon rcicon rcicon-check"
      >
        
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
      <div
        class="rc-steps-item-description"
      >
        xx
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps render renders step with tailContent 1`] = `
<div
  class="rc-steps rc-steps-horizontal rc-steps-label-horizontal"
>
  <div
    class="rc-steps-item rc-steps-item-process"
  >
    <div
      class="rc-steps-item-tail"
    >
      text
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        1
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        已完成
      </div>
      <div
        class="rc-steps-item-description"
      >
        xx
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      <div>
        content
      </div>
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        2
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        进行中
      </div>
      <div
        class="rc-steps-item-description"
      >
        xx
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      3
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        3
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
      <div
        class="rc-steps-item-description"
      >
        xx
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      text
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        4
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
      <div
        class="rc-steps-item-description"
      >
        xx
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps render renders vertical correctly 1`] = `
<div
  class="rc-steps rc-steps-vertical"
>
  <div
    class="rc-steps-item rc-steps-item-process"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        1
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        已完成
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        2
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        进行中
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        3
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        4
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps render renders with fasly children 1`] = `
<div
  class="rc-steps rc-steps-horizontal rc-steps-label-horizontal"
>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        1
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        已完成
      </div>
      <div
        class="rc-steps-item-description"
      >
        xx
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        2
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        进行中
      </div>
      <div
        class="rc-steps-item-description"
      >
        xx
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-process"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        3
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
      <div
        class="rc-steps-item-description"
      >
        xx
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-finish"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon rcicon rcicon-check"
      >
        
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
      <div
        class="rc-steps-item-description"
      >
        xx
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps render renders step with specified starting number 1`] = `
<div
  class="rc-steps rc-steps-horizontal rc-steps-label-horizontal"
>
  <div
    class="rc-steps-item rc-steps-item-process"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        4
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        已完成
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        5
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        进行中
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        6
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
    </div>
  </div>
  <div
    class="rc-steps-item rc-steps-item-wait"
  >
    <div
      class="rc-steps-item-tail"
    >
      
    </div>
    <div
      class="rc-steps-item-icon"
    >
      <span
        class="rc-steps-icon"
      >
        7
      </span>
    </div>
    <div
      class="rc-steps-item-content"
    >
      <div
        class="rc-steps-item-title"
      >
        待运行
      </div>
    </div>
  </div>
</div>
`;
